ethereum/core/src/main/java/tech/pegasys/teku/core/epoch/status/ValidatorStatuses.java

```diff
@@ -0,0 +1,139 @@
+
+    return new ValidatorStatuses(statuses, totalBalances);
+  }
+
+  private static void processAttestations(
+      final List<ValidatorStatus> statuses,
+      final BeaconState state,
+      final UInt64 previousEpoch,
+      final UInt64 currentEpoch) {
+    Stream.concat(
+            state.getPrevious_epoch_attestations().stream(),
+            state.getCurrent_epoch_attestations().stream())
+        .forEach(
+            attestation -> {
+              final AttestationData data = attestation.getData();
+
+              final AttestationUpdates updates = new AttestationUpdates();
+              final Checkpoint target = data.getTarget();
+              if (target.getEpoch().equals(currentEpoch)) {
+                updates.currentEpochAttester = true;
+                updates.currentEpochTargetAttester =
+                    matchesEpochStartBlock(state, currentEpoch, target.getRoot());
+              } else if (target.getEpoch().equals(previousEpoch)) {
+                updates.previousEpochAttester = true;
+
+                updates.inclusionInfo =
+                    Optional.of(
+                        new InclusionInfo(
+                            attestation.getInclusion_delay(), attestation.getProposer_index()));
+
+                if (matchesEpochStartBlock(state, previousEpoch, target.getRoot())) {
+                  updates.previousEpochTargetAttester = true;
+
+                  updates.previousEpochHeadAttester =
+                      get_block_root_at_slot(state, data.getSlot())
+                          .equals(data.getBeacon_block_root());
+                }
+              }
+
+              // Apply flags to attestingIndices
+              AttestationUtil.stream_attesting_indices(
+                      state, data, attestation.getAggregation_bits())
+                  .mapToObj(statuses::get)
+                  .forEach(updates::apply);
+            });
+  }
+
+  private static boolean matchesEpochStartBlock(
+      final BeaconState state, final UInt64 currentEpoch, final Bytes32 root) {
+    return BeaconStateUtil.get_block_root(state, currentEpoch).equals(root);
+  }
+
+  public TotalBalances getTotalBalances() {
+    return totalBalances;
+  }
+
+  public List<ValidatorStatus> getStatuses() {
+    return statuses;
+  }
+
+  public int getValidatorCount() {
+    return statuses.size();
+  }
+
+  private static class AttestationUpdates {
+    private boolean currentEpochAttester = false;
+    private boolean currentEpochTargetAttester = false;
+    private boolean previousEpochAttester = false;
+    private boolean previousEpochTargetAttester = false;
+    private boolean previousEpochHeadAttester = false;
+    private Optional<InclusionInfo> inclusionInfo = Optional.empty();
+
+    public void apply(final ValidatorStatus status) {
+      status.updateCurrentEpochAttester(currentEpochAttester);
+      status.updateCurrentEpochAttester(currentEpochAttester);
```

Comment:
is the double call here intentional?

Reply:
No, that would explain how I originally managed to miss calling `updatePreviousEpochAttester` though...